### ISSUE TITLE
Make GitHub token inputs/behaviour work like other ecosystem actions

### ISSUE THREAD

**Issue by jcbhmr** [spenserblack/actions-wiki#59]
basically like this:

![image](https://user-images.githubusercontent.com/61068799/231873180-10407c85-6c45-4734-b8e6-01912a6da4d8.png)
https://github.com/actions/checkout

heck, we can even just _literally use the actions/checkout action_ since this is a _composite_ action to clone the .wiki.git folder into some /tmp/myfolder and then copy stuff over to it and then push _all without touching the github token_ (since it saves it in the remoteurl thing when it clones it, right?)

**Comment by jcbhmr** [comment]
i recommend closing #2 in favor of this since i have no idea what that issue was originally targetting. if you @spenserblack could fill me in thatd be great 😊

**Comment by spenserblack** [comment]
I'm not really sure what the suggested change is (besides using `actions/checkout` in this action). How would the behavior change?

> maybe close #2 in favor of this?

For #2, "support" isn't just code, but documentation. So a simple documentation change mentioning fine-grained tokens, and the differences (if any) between the two would resolve #2 for me.

**Comment by jcbhmr** [comment]
> I'm not really sure what the suggested change is (besides using `actions/checkout` in this action). How would the behavior change?

specifically:

```yml
- uses: spenserblack/actions-wiki@v1
  with:
    # No need for a token: ... setting
```

```yml
- uses: spenserblack/actions-wiki@v1
  with:
    # If you DO use a token: ... setting, it's without a username (no more x:gh_abc123...)
    token: ${{ secrets.GITHUB_TOKEN }}
```

```yml
- uses: spenserblack/actions-wiki@v1
  with:
    # If you use a custom PAT, you also don't need a username
    token: ${{ secrets.OCTOCAT_GITHUB_PAT }}
```

**Comment by jcbhmr** [comment]
> > maybe close #2 in favor of this?
> 
> For #2, "support" isn't just code, but documentation. So a simple documentation change mentioning fine-grained tokens, and the differences (if any) between the two would resolve #2 for me.

i think they all literally work the exact same. at least for the github api: https://docs.github.com/en/rest/overview/authenticating-to-the-rest-api?apiVersion=2022-11-28

for the `git clone https://user:pass@github.com/octocat/project` i don't know? it sounds like you can just delegate to a token using `gh auth login` or something

official docs: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github#https

basically what i'm getting at is that it's all just a bunch of tokens, right? and i THINK you can just use `git clone https://x:$TOKEN@github.com/user/repo.wiki.git`

if a user encounters an issue with an oauth2 token, i think that's a THEN problem, not a NOW problem. right now i just reallyreallyreally want to get rid of the blasted `token: x:${{ secrets.GITHUB_TOKEN }}` that's required. that and make it work like literally every other github action dealing with tokens 😆 where you just pass the token, no username (hence the idea to just delegate to another action like actions/checkout)

**Comment by spenserblack** [comment]
OK, yeah, I'm 100% on board with not requiring a username in the `token` input. The username could potentially be an input, with a default value so it's not required by default. I'll keep #2 open, and put it in the icebox.

**Comment by jcbhmr** [comment]
I'm still a bit confused about the need for a username: input. Can we simply use a token: without it? A couple of things to consider:

The username: field seems unnecessary for authentication from the command line.
I couldn't find any similar tools that require a username: input, such as https://github.com/actions/checkout or https://github.com/ad-m/github-push-action, which only require a token. Even the gh CLI tool doesn't seem to need a username for any of its git CLI operations!

@spenserblack Is there some other scenario that I'm missing or misunderstanding?

**Comment by spenserblack** [comment]
Frankly, I'm confused why `user:TOKEN@github.com` works for cloning wikis, but not `TOKEN@github.com`, too. Don't really have an explanation why adding an arbitrary `x:` makes cloning wikis work, which is why I was so convinced that a custom PAT was required for a while. I wonder if it should be raised in the GitHub community discussions as a bug :thinking:

I've never even tried it, but I wonder if `actions/checkout` would be able to clone a wiki without a custom PAT. And if so, what are they doing differently from this action?

**Comment by jcbhmr** [comment]
> Frankly, I'm confused why `user:TOKEN@github.com` works for cloning wikis, but not `TOKEN@github.com`, too. Don't really have an explanation why adding an arbitrary `x:` makes cloning wikis work, which is why I was so convinced that a custom PAT was required for a while. I wonder if it should be raised in the GitHub community discussions as a bug 🤔

if I had to take a guess, id say that it has something to do with the different visibility "guaranteed" or expectations of username vs passwords. cause:

```sh
https://token@github.com
        ^^^^^ username!
https://user:token@github.com
             ^^^^^ now it's the "password" slot!
```

like you know how passwords get `Password: *******` when you paste them in when doing `git push` when you're not authed for the first time? might also have to do with the credential manager? like maybe it stores user/hostname pairs and then expects the user to enter the password at push time? idk 🤷

**Comment by spenserblack** [comment]
That's a good point about how `foo:<secret>` puts the secret in the dedicated password slot.
TBH I wonder if this is officially supported or an "undocumented feature," since I haven't seen the docs mention it anywhere.

**Comment by jcbhmr** [comment]
already done in https://github.com/spenserblack/actions-wiki/pull/65 lmao
